executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Chain.hs

```diff
@@ -113,105 +110,99 @@ instance ( HashAlgorithm hashAlgo
-                 , transactionsSt = mempty
-                 }
+       IRC Env { headerEnv = Header.Env{ Header.initialSlot = initsl}
+               } <- judgmentContext
+       pure $! St { headerSt
+                      = Header.St { Header.currentSlot = initsl
+                                  , Header.hupdateSt
+                                      = Hupdate.St { Hupdate.wrsips
+                                                      = Map.empty
+                                                   , Hupdate.asips
+                                                      = Map.empty
+                                                   }
+                                  }
+                  , bodySt
+                      = Body.St
+                        { Body.wrsips = Map.empty
+                        , Body.transactionSt
+                            = Transaction.St
+                                 { Transaction.wrsips = Map.empty
+                                 , Transaction.utxoSt = mempty
+                                 , Transaction.updateSt
+                                      = Update.St { Update.wrsips = Map.empty
+                                                  , Update.ideationSt
+                                                      = Ideation.St
+                                                        { Ideation.subsips
+                                                            = Set.empty
+                                                        , Ideation.wssips
+                                                            = Map.empty
+                                                        , Ideation.wrsips
+                                                            = Map.empty
+                                                        , Ideation.ballots
+                                                            = Map.empty
+                                                        , Ideation.voteResultSIPs
+                                                            = Map.empty
+                                                        }
+                                                  , Update.implementationSt
+                                                      = Implementation.St ()
+                                                  }
+                                 }
+                        }
+                  }
     ]
 
   transitionRules = [
     do
-      TRC ( Env { maximumBlockSize, transactionsEnv }
-          , St  { currentSlot
-                , openVotingPeriods
-                , closedVotingPeriods
-                , transactionsSt =
-                    Transaction.St { -- Transaction.openVotingPeriods
-                                     Transaction.utxoSt
-                                   , Transaction.updateSt
-                                   }
+      TRC ( Env { maximumBlockSize, headerEnv, bodyEnv = Body.Env{Body.transactionEnv} }
+          , St  { headerSt
```

Comment:
To discuss: I'm thinking whether it'd make sense to unwrap the header state, so that it is clearer which components we're passing to the header rule. Ideally we wouldn't do this since it'd break abstraction, but on the other hand we're unwrapping this state anyway to pass it to the body rule, so we might want to be explicit about it for the sake of symmetry (otherwise we see `headerSt` in the pre-state of the transition, but `Header.St { .... }` in the post-state) :man_shrugging: